src/main.rs

```diff
@@ -190,40 +196,51 @@ fn path_name(path: &Path) -> &'static str {
     }
 }
 
+fn normalize_path(request_path: &str) -> (&str, &str) {
+    if let Some(trimmed_path) = request_path.strip_prefix("/api") {
+        if let Some(maybe_api_version) = trimmed_path.split('/').nth(1) {
+            if let Some(version_number) = maybe_api_version.strip_prefix('v') {
+                if version_number.parse::<u8>().is_ok() {
+                    // 6 = /api/v
+                    let len = 6 + version_number.len();
+                    return (&request_path[..len], &request_path[len..]);
+                };
+            };
+        }
+
+        ("/api", trimmed_path)
+    } else {
+        ("/api", request_path)
+    }
+}
+
 async fn handle_request(
```

Comment:
It would be nice if we could split things out from this function so they could be unit tested